x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/conditional/Case.java

```diff
@@ -37,8 +43,12 @@
 
 import static org.elasticsearch.common.logging.LoggerMessageFormat.format;
 import static org.elasticsearch.xpack.esql.core.type.DataType.NULL;
+import static org.elasticsearch.xpack.esql.io.stream.PlanNameRegistry.PlanReader.readerFromPlanReader;
+import static org.elasticsearch.xpack.esql.io.stream.PlanNameRegistry.PlanWriter.writerFromPlanWriter;
 
 public final class Case extends EsqlScalarFunction {
+    public static final NamedWriteableRegistry.Entry ENTRY = new NamedWriteableRegistry.Entry(Expression.class, "Case", Case::new);
```

Comment:
I'm intentionall not using `Class.class.simpleName()` for this because renaming the class would change the serialization. I don't think we'll rename the classes, but I don't want to accidentally change things.

Reply:
Ah, thanks, that was the reason to [this comment](https://github.com/elastic/elasticsearch/pull/109708#discussion_r1644811067).
That makes sense, though it opens the possibility for other (c&p) issues. And on class renaming, the string can be hard coded.
But both low probabilities and since this seems an established practice, 👍.

Reply:
Yeah. It's more normal to use a string here. I like how it's a little paranoid about future mistakes.